Classes/Hooks/FileDumpHook.php

```diff
@@ -176,10 +177,13 @@ public function checkFileAccess(ResourceInterface $file)
             }
         }
 
+        // Dump the precise requested file for File and ProcessedFile, but dump the referenced file for FileReference
+        $dumpFile = $file instanceof FileReference ? $file->getOriginalFile() : $file;
+
         if ($this->forceDownload($this->originalFile->getExtension())) {
```

Comment:
This should also check the dumpFile

Reply:
What do you mean? Can you elaborate?

Reply:
`if ($this->forceDownload($dumpFile->getExtension())) {`